src/component/providers/AccessProvider/AccessProvider.tsx

```diff
@@ -1,88 +1,110 @@
-import { FC } from 'react';
-
-import AccessContext from '../../../contexts/AccessContext';
-import useUser from '../../../hooks/api/getters/useUser/useUser';
+import { ReactElement, ReactNode, useCallback, useMemo } from 'react';
+import AccessContext, { IAccessContext } from '../../../contexts/AccessContext';
+import { useAuth } from '../../../hooks/api/getters/useAuth/useAuth';
 import { ADMIN } from './permissions';
+import { IPermission } from '../../../interfaces/user';
 
-// TODO: Type up redux store
-interface IAccessProvider {
-    store: any;
-}
-
-interface IPermission {
-    permission: string;
-    project?: string | null;
-    environment: string | null;
+interface IAccessProviderProps {
+    children: ReactNode;
+    permissions: IPermission[];
 }
 
-const AccessProvider: FC<IAccessProvider> = ({ store, children }) => {
-    const { permissions } = useUser();
-    const isAdminHigherOrder = () => {
-        let called = false;
-        let result = false;
-
-        return () => {
-            if (called) return result;
-            const permissions = store.getState().user.get('permissions') || [];
-            result = permissions.some(
-                (p: IPermission) => p.permission === ADMIN
+const AccessProvider = (props: IAccessProviderProps): ReactElement => {
+    const { auth } = useAuth();
+    const permissions = props.permissions ?? auth?.permissions;
```

Comment:
Is there any benefit of receiving this from props rather than useAuth? I imagine the source would be the same for both of these.

Reply:
Yeah, the prop is only used for testing. Mocking `useAuth` would probably be better, but I think that should be its own PR.

Reply:
Makes sense! Maybe add a comment so we remember?